hypothesis-python/src/hypothesis/stateful.py

```diff
@@ -73,52 +70,61 @@ def __delete__(self, obj):
         raise AttributeError(u'Cannot delete TestCase')
 
 
-def find_breaking_runner(state_machine_factory, settings=None):
-    def is_breaking_run(runner):
-        try:
-            runner.run(state_machine_factory())
-            return False
-        except HypothesisException:
-            raise
-        except EXCEPTIONS_TO_FAIL:
-            verbose_report(traceback.format_exc)
-            return True
+def run_state_machine_as_test(state_machine_factory, settings=None):
+    """Run a state machine definition as a test, either silently doing nothing
+    or printing a minimal breaking program and raising an exception.
+
+    state_machine_factory is anything which returns an instance of
+    GenericStateMachine when called with no arguments - it can be a class or a
+    function. settings will be used to control the execution of the test.
+    """
     if settings is None:
         try:
             settings = state_machine_factory.TestCase.settings
+            assert isinstance(settings, Settings)
         except AttributeError:
             settings = Settings.default
 
-    search_strategy = StateMachineSearchStrategy(settings)
+    @settings
+    @given(st.data())
+    def run_state_machine(machine, data):
+        state_machine = machine()
+        data.data.hypothesis_runner = state_machine
```

Comment:
If we're going to be using this field I think we should rename it to something a bit more explicit. `conjecture_data` maybe?